File: examples/leetcode-out/swift/23/merge-k-sorted-lists.swift
import Foundation

func mergeKLists(_ lists: [[Int]]) -> [Int] {
	let lists = lists
	
	let k = lists.count
	var indices: [Int] = []
	var i = 0
	while i < k {
		indices = indices + [0]
		i = i + 1
	}
	var result: [Int] = []
	while true {
		var best = 0
		var bestList = -1
		var found = false
		var j = 0
		while j < k {
			let idx = indices[j]
			if idx < lists[j].count {
				let val = lists[j][idx]
				if !found || val < best {
					best = val
					bestList = j
					found = true
				}
			}
			j = j + 1
		}
		if !found {
			break
		}
		result = result + [best]
		indices[bestList] = indices[bestList] + 1
	}
	return result
}

func main() {
}
main()

File: examples/leetcode-out/dart/23/merge-k-sorted-lists.dart
dynamic mergeKLists(lists) {
	dynamic k = lists.length;
	dynamic indices = [];
	dynamic i = 0;
	while ((i < k)) {
		indices = (indices + [0]);
		i = ((i + 1)).toInt();
	}
	dynamic result = [];
	while (true) {
		dynamic best = 0;
		dynamic bestList = -1;
		dynamic found = false;
		dynamic j = 0;
		while ((j < k)) {
			dynamic idx = indices[j];
			if ((idx < lists[j].length)) {
				dynamic val = lists[j][idx];
				if ((!found || (val < best))) {
					best = val;
					bestList = j;
					found = true;
				}
			}
			j = ((j + 1)).toInt();
		}
		if (!found) {
			break;
		}
		result = (result + [best]);
		indices[bestList] = (indices[bestList] + 1);
	}
	return result;
}

void main() {
}

